cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobDetail.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.runtimejob;
+
+import io.cdap.cdap.runtime.spi.ProgramRunInfo;
+
+import java.util.Objects;
+
+/**
+ * Status details of dataproc runtime job.
+ */
+public class DataprocRuntimeJobDetail extends RuntimeJobDetail {
+
+  private final String jobStatusDetails;
+
+  public DataprocRuntimeJobDetail(ProgramRunInfo runInfo, RuntimeJobStatus status, String jobStatusDetails) {
+    super(runInfo, status);
+    this.jobStatusDetails = jobStatusDetails;
+  }
+
+  /**
+   * Returns string representation of dataproc job status details.
+   */
+  public String getJobStatusDetails() {
+    return jobStatusDetails;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    DataprocRuntimeJobDetail that = (DataprocRuntimeJobDetail) o;
+    return runInfo.equals(runInfo)
```

Comment:
That's not a correct implementation of equals. You should be comparing runInfo with that.runInfo. and same for the status in the next line.

Reply:
Fixed.

Reply:
For status and runInfo, should you not use the equals implementation from RuntimeJobDetails? It already implements equals correctly.

Reply:
The `jobStatusDetails` field also needs to be compared in addition the fields compared in `RuntimeJobDetails`. We can reuse the equals method of `RuntimeJobDetails `, but it should not make much difference.

Reply:
jobStatusDetails should be compared here, because this class introduces it. The base class already correctly implemented equals. If the base class adds a new field, you will have to change this equals implementation as well. The right approach is to reuse.